canon/src/repr_hosted.rs

```diff
@@ -90,16 +90,18 @@ where
         // can we inline the value?
         if len <= buffer.as_ref().len() {
             let mut sink = ByteSink::new(buffer.as_mut(), store.clone());
-            t.write(&mut sink)?;
+            // TODO: Handle this as a host error
+            t.write(&mut sink).unwrap();
```

Comment:
Its unclear to me how the errors are gonna be passed to the host if this I/O fails

Reply:
"In the hosted case, we can assume that our allocation works, and throw any potential error as a panic for he host."

This is writing to bytes in memory, it's not std::io.

Reply:
ie, we panic instead of dealing with the error in the hosted environment

Reply:
It's ok to keep as is for the time being but the comment should be more exhaustive. It's unclear what are the ramification of having the `unwrap` temporary there, and what would be the possible approach to handle as host error. Also, I would prefer `expect` instead of `unwrap` if it's possible, since it will provide a bit more context.